.github/workflows/lint-changed-files.yaml

```diff
@@ -0,0 +1,44 @@
+# Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
```

Comment:
Could you pull `lint-changed-files.yaml` and `.lintr` from `packagetemplate` please? I added some extra checks

Reply:
Added in https://github.com/epiverse-trace/datadelay/pull/11/commits/a68473618a3e77408c56e3f9ec4011975c58f000.

Just a heads up that the R version check in the `.lintr` file does not play nice with the DESCRIPTION entry `Depends: R version (>= 2.10)`. This is added by `usethis::use_data()`, even if `version = 3` is specified. I've manually set the R version dependency to `>= 3.5.0` in https://github.com/epiverse-trace/datadelay/pull/11/commits/f75cf84f84ef390fc3459e3b27782f0dc8d0b187.